src/app/tests/TestWriteInteraction.cpp

```diff
@@ -0,0 +1,314 @@
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    app::WriteClient writeClient;
+
+    chip::app::InteractionModelDelegate delegate;
+    System::PacketBufferHandle buf = System::PacketBufferHandle::New(System::PacketBuffer::kMaxSize);
+    err                            = writeClient.Init(&gExchangeManager, &delegate, 0);
+    NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
+    AddAttributeDataElement(apSuite, apContext, writeClient);
+
+    err = writeClient.SendWriteRequest(kTestDeviceNodeId, gAdminId, nullptr);
+    NL_TEST_ASSERT(apSuite, err == CHIP_ERROR_NOT_CONNECTED);
+
+    GenerateWriteResponse(apSuite, apContext, buf);
+
+    err = writeClient.ProcessWriteResponseMessage(std::move(buf));
+    NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
+
+    writeClient.Shutdown();
+}
+
+void TestWriteInteraction::TestWriteHandler(nlTestSuite * apSuite, void * apContext)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    app::WriteHandler writeHandler;
+
+    chip::app::InteractionModelDelegate IMdelegate;
+    System::PacketBufferHandle buf = System::PacketBufferHandle::New(System::PacketBuffer::kMaxSize);
+    err                            = writeHandler.Init(&IMdelegate);
+
+    GenerateWriteRequest(apSuite, apContext, buf);
+
+    err = writeHandler.ProcessWriteRequest(std::move(buf));
+    NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
+
+    AddAttributeStatus(apSuite, apContext, writeHandler);
+
+    writeHandler.mpExchangeCtx = gExchangeManager.NewContext({ 0, 0, 0 }, nullptr);
+    TestExchangeDelegate delegate;
+    writeHandler.mpExchangeCtx->SetDelegate(&delegate);
+    err = writeHandler.SendWriteResponse();
+    NL_TEST_ASSERT(apSuite, err == CHIP_ERROR_NOT_CONNECTED);
+
+    writeHandler.Shutdown();
+}
+} // namespace app
+} // namespace chip
+
+namespace {
+
+void InitializeChip(nlTestSuite * apSuite)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    chip::Optional<chip::Transport::PeerAddress> peer(chip::Transport::Type::kUndefined);
+    chip::Transport::AdminPairingTable admins;
+    chip::Transport::AdminPairingInfo * adminInfo = admins.AssignAdminId(chip::gAdminId, chip::kTestDeviceNodeId);
+
+    NL_TEST_ASSERT(apSuite, adminInfo != nullptr);
+
+    err = chip::Platform::MemoryInit();
+    NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
+
+    chip::gSystemLayer.Init(nullptr);
+
+    err = chip::gSessionManager.Init(chip::kTestDeviceNodeId, &chip::gSystemLayer, &chip::gTransportManager, &admins,
+                                     &chip::gMessageCounterManager);
+    NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
+
+    err = chip::gExchangeManager.Init(&chip::gSessionManager);
+    NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
+
+    err = chip::gMessageCounterManager.Init(&chip::gExchangeManager);
+    NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
+}
+
+/**
+ *   Test Suite. It lists all the test functions.
+ */
+
+// clang-format off
+    const nlTest sTests[] =
```

Comment:
indent here seems off - need to manually correct since clang-format is off.

Reply:
Done, thanks